canvas_modules/common-canvas/src/common-properties/properties-controller.js

```diff
@@ -227,6 +227,42 @@ export default class PropertiesController {
 		return this.appData;
 	}
 
+	getActiveTabPropertyValues() {
+		const activeTab = this.propertiesStore.getActiveTab();
+		const allPropertyValues = this.propertiesStore.getPropertyValues();
+		const activeTabPropertyValues = {};
+		let activeTabPropertyIds = [];
+
+		for (const group of this.form.groupMetadata.groups) {
+			if (group.name === activeTab) {
+				activeTabPropertyIds = this.getGroupProperties(group);
+				break;
+			}
+		}
+
+		activeTabPropertyIds.forEach((propertyId) => {
+			activeTabPropertyValues[propertyId] = allPropertyValues[propertyId];
+		});
+
+		return activeTabPropertyValues;
+	}
+
+	// Recursively get all properties under given uiGroup
+	getGroupProperties(uiGroup) {
```

Comment:
Have you looked at `this.panelTree`?  I think this already has a tree of all the panels and controls within a tree.  Could we use that instead of having to parse it again?

Reply:
@matthoward366 `this.panelTree` works for most cases. But when `"type": "subTabs",`, the group doesn't show up under `this.panelTree`.  

Example in `tab_paramDef.json`, `Primary3` group has `"type": "subTabs",`. This group isn't listed under panelTree - https://github.com/elyra-ai/canvas/blob/main/canvas_modules/harness/test_resources/parameterDefs/tab_paramDef.json#L577

<img width="845" alt="Screenshot 2023-11-09 at 6 14 15 AM" src="https://github.com/elyra-ai/canvas/assets/25124000/7502bab2-54f4-43a6-9e4d-4609428497da">

Reply:
Should it be?  Maybe the code for the panelTree parsing isn't handling this correctly?

Reply:
The other question here is should document the method in the controller for getting the active tab then the consuming application should be able to do this to determine the parameters within that group.